parser.go

```diff
@@ -262,10 +264,20 @@ func (p *parser) check() (Result, string, unused, error) {
 			p.fireMatch(token, result, s, ttl, err)
 			return result, s, unused{mechanisms[i+1:], redirect}, err
 		}
+
+		// Store the first match result if not already set
+		if p.ignoreMatches && matches && p.firstMatchResult == None {
+			p.firstMatchResult = result
+		}
```

Comment:
Why not also fireMatch()? as in L260:L264

Reply:
The purpose of fireMatch is to notify a listener when a mechanism matches, and in this case, we are deliberately choosing to ignore matches, so it wouldn't make sense to call to fireMatch() from my opinion. We are only storing the firstMatch for later reference .